lib/src/test/java/io/ably/lib/test/rest/RestAuthTest.java

```diff
@@ -79,6 +79,23 @@ public void authinit0() {
 	}
 
 	/**
+	 * Init library with useTokenAuth set
+	 */
+	@Test
+	public void authinit0point5() {
+		try {
+			TestVars testVars = Setup.getTestVars();
+			ClientOptions opts = new ClientOptions(testVars.keys[0].keyStr);
+			opts.useTokenAuth = true;
+			AblyRest ably = new AblyRest(opts);
+			assertEquals("Unexpected Auth method mismatch", ably.auth.getAuthMethod(), AuthMethod.token);
```

Comment:
Why not simply check if a token exists after a REST operation i.e. inspect the Auth object to ensure it has not token, perform a REST operation, inspect the Auth object to see if it now has a token.  Should be pretty straightforward.